src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.WithFrozenSourceGeneratedDocumentsCompilationTracker.cs

```diff
@@ -49,6 +49,9 @@ public WithFrozenSourceGeneratedDocumentsCompilationTracker(
             ICompilationTracker underlyingTracker,
             TextDocumentStates<SourceGeneratedDocumentState> replacementDocumentStates)
         {
+            // We should never create a chain of trackers wrapping each other.
+            Contract.ThrowIfTrue(underlyingTracker is WithFrozenSourceGeneratedDocumentsCompilationTracker);
```

Comment:
thanks!

Reply:
another, possibly better, way to do this would be to change this to take a RegularCompilationTracker.

Reply:
Some nice man on the Roslyn team did this for me, so I've removed the contract :)